rpmlint/checks/SystemdTmpfilesCheck.py

```diff
@@ -0,0 +1,270 @@
+            path = Path(location)
+            if not path.is_absolute():
+                raise Exception(f'Absolute path expected: {path}')
+            node = self.dropin_dir_trie
+            # Skip initial '/'.
+            parts = path.parts[1:]
+            for part in parts[:-1]:
+                node = node.setdefault(part, {})
+                if node is None:
+                    raise Exception(f'Conflicting paths in trie {location}')
+            node[parts[-1]] = None
+
+    def _is_path_restricted(self, pkgfile):
+        """Returns True if the given path is within a tmpfiles.d drop in
+        directory.
+        """
+        if stat.S_ISDIR(pkgfile.mode):
+            # we only care for configuration files
+            return False
+        elif not pkgfile.name.endswith('.conf'):
+            # systemd-tmpfiles only considers files ending in *.conf
+            return False
+
+        path = Path(pkgfile.name)
+
+        # now check the trie if the path is in a restricted location
+
+        # Skip initial '/'
+        parts = path.parts[1:]
+        node = self.dropin_dir_trie
+        for part in parts:
+            if node is None:
+                return True
+            if part not in node:
+                return False
+            node = node[part]
+        return False
+
+    def _normalize_whitelist_entry(self, entry):
+        """Perform any operations on the whitelisting entry to make it match
+        the format excepted by the rest of the checker.
+        """
+        packages = entry.setdefault('packages', [])
+        if 'package' in entry:
+            # _sanity_check_whitelist_entry() makes sure only one of both keys
+            # is present in an entry
+            packages.append(entry['package'])
+
+    def _sanity_check_whitelist_entry(self, entry):
+        package = entry.get('package', None)
+        packages = entry.get('packages', [])
+
+        if not package and not packages:
+            raise KeyError('SystemdTmpfilesCheck: missing "package" or "packages" key in whitelisting entry')
+        elif package and packages:
+            raise KeyError('SystemdTmpfilesCheck: encountered both "package" and "packages" keys in FileDigestGroup entry')
+
+        if package and not isinstance(package, str):
+            raise KeyError('SystemdTmpfilesCheck: "package" key contains non-string value')
+        elif packages and not isinstance(packages, list):
+            raise KeyError('SystemdTmpfilesCheck: "packages" key contains non-list value')
+
+        if 'path' not in entry:
+            raise KeyError('SystemdTmpfilesCheck: missing "path" key')
+
+    def _is_in_safe_location(self, tf_entry):
+        """Returns whether the given tmpfiles configuration entry affects a
+        safe location that is usually okay to operate in."""
+        # don't add tmp directories here, they could still be dangerous, since
+        # publicly accessible
+        WELL_KNOWN = (
```

Comment:
Please move this to the beginning of the file, similarly as we do for other checks.